server/stycoon/history.go

```diff
@@ -0,0 +1,81 @@
+package stycoon
+
+import (
+	"database/sql"
+	"encoding/json"
+)
+
+func (game *Game) reportStaticData() error {
+	_, err := game.db.Exec("insert into d_history (`season`, `tick`, `static-data`) values (?, ?, ?)", game.Tick.Season, -1, game.SerializedStaticData)
+	return err
+}
+
+func (game *Game) report() error {
+	// TODO report report
+	reportUserId := int64(-1)
+	data, err := game.GetData(&reportUserId)
+	if err != nil {
+		return err
+	}
+	serializedData, err := json.Marshal(data)
+	if err != nil {
+		return err
+	}
+	_, err = game.db.Exec("insert into d_history (`season`, `tick`, `data`) values (?, ?, ?)", game.Tick.Season, game.Tick.Tick, serializedData)
+	return err
+}
+
+func FilterCommands(playerId *int64, ships *map[string]ShipsValue) {
+	for k, ship := range *ships {
+		if ship.Command == nil {
+			continue
+		}
+		shouldRemove := true
+		if playerId != nil && *playerId == ship.Player {
+			shouldRemove = false
+		}
+		if !shouldRemove {
+			continue
+		}
+		filteredShip := ship
+		filteredShip.Command = nil
+		(*ships)[k] = filteredShip
+	}
+}
+
+func (game *Game) History(id HistoryIdentifier, playerId *int64) (HistoryEntry, error) {
+	var entry HistoryEntry
+	var staticData sql.NullString
+	err := game.db.QueryRow("select `static-data` from d_history where `season` = ? and `tick` = -1", id.Season).Scan(&staticData)
```

Comment:
oh god i see it now. please split it into two tables instead.

Reply:
one table for static data and second table for data and reports

Reply:
Ok, first I thought that this is simpler, but it is exactly the same (apart from lower readability). Changed